35-CAN-vehicle-monitor/firmware/README.md

```diff
@@ -0,0 +1,147 @@
+
+![MCU and components, not soldered](../images/mcu_not_soldered.jpg)
+
+Adafruit has a [handy guide for soldering headers](https://learn.adafruit.com/how-to-solder-headers/overview). Once you've soldered on the headers and the terminal block, your MCU should look like this:
+
+![MCU and components, soldered](../images/mcu_soldered.jpg)
+
+Now, plug the headers of the MCU into the female headers on the Notecarrier F. Install the Notecard into the Notecarrier and the antenna onto the Notecard as described in the [Notecard Quickstart guide](https://dev.blues.io/quickstart/notecard-quickstart/notecard-and-notecarrier-f/#connect-your-notecard-and-notecarrier). Finally, connect the MCU to your development PC with a USB C cable.
+
+### USB to CAN Converter
+
+The USB to CAN converter (the "converter") allows us to send CAN messages from our development PC to the MCU via a CAN bus. A CAN bus requires two signal lines: CAN high (CANH) and CAN low (CANL). The converter comes in two parts. Plug the part with the USB connector into your development PC. Connect the second part to the first part via the RS232 connector. This second part has a screw terminal block with ports labeled CANH and CANL. Using a Philips head screwdriver, loosen the screws above these two connectors, insert the end of a male-to-male jumper wire into each, and tighten the screws, securing the wires. Similarly, take the free ends of each jumper wire and plug the CANL wire into the port labeled L on the MCU's terminal block and the CANH wire into the port labeled H. At this point, the hardware is fully set up and should look like this:
+
+![Complete hardware](../images/complete_hardware.jpg)
+
+## Notehub Setup
+
+Sign up for a free account on [notehub.io](https://notehub.io) and [create a new project](https://dev.blues.io/quickstart/notecard-quickstart/notecard-and-notecarrier-pi/#set-up-notehub).
+
+## Firmware Setup
+
+To build and upload the firmware onto the MCU, you'll need VSCode with the PlatformIO extension.
+
+1. Download and install [Visual Studio Code](https://code.visualstudio.com/).
+1. Install the [PlatformIO IDE extension](https://marketplace.visualstudio.com/items?itemName=platformio.platformio-ide) via the Extensions menu of Visual Studio Code.
+1. Click the PlatformIO icon on the left side of VSCode, then click Pick a folder, and select the the firmware directory, `35-CAN-vehicle-monitor/firmware`.
+1. In the file explorer, open main.cpp and uncomment this line: `// #define PRODUCT_UID "com.my-company.my-name:my-project"`. Replace `com.my-company.my-name:my-project` with the ProductUID of the Notehub project you created in [Notehub Setup](#notehub-setup).
```

Comment:
```suggestion
1. In the file explorer, open `main.cpp` and uncomment this line: `// #define PRODUCT_UID "com.my-company.my-name:my-project"`. Replace `com.my-company.my-name:my-project` with the ProductUID of the Notehub project you created in [Notehub Setup](#notehub-setup).
```

Reply:
You might want to link to this section https://dev.blues.io/notehub/notehub-walkthrough/#finding-a-productuid.